src/Glpi/DependencyInjection/PluginContainer.php

```diff
@@ -0,0 +1,198 @@
+        $this->kernel = $kernel;
+    }
+
+    public function set(string $id, ?object $service): void
+    {
+        $this->internal_container->set($id, $service);
+    }
+
+    public function initialized(string $id): bool
+    {
+        return $this->internal_container->initialized($id);
+    }
+
+    public function getParameter(string $name): \UnitEnum|float|array|bool|int|string|null
+    {
+        return $this->internal_container->getParameter($name) ?? $this->symfony_container->getParameter($name);
+    }
+
+    public function hasParameter(string $name): bool
+    {
+        return $this->internal_container->hasParameter($name) || $this->symfony_container->hasParameter($name);
+    }
+
+    public function setParameter(string $name, \UnitEnum|float|array|bool|int|string|null $value): void
+    {
+        $this->internal_container->setParameter($name, $value);
+    }
+
+    public function get(string $id, int $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE): ?object
+    {
+        $this->initializeContainer();
+
+        try {
+            return $this->internal_container->get($id, self::EXCEPTION_ON_INVALID_REFERENCE);
+        } catch (NotFoundExceptionInterface) {
+        }
+
+        return $this->symfony_container->get($id, $invalidBehavior);
+    }
+
+    public function has(string $id): bool
+    {
+        $this->initializeContainer();
+
+        if ($this->internal_container->has($id)) {
+            return true;
+        }
+
+        return $this->symfony_container->has($id);
+    }
+
+    public function initializeContainer(): void
+    {
+        if ($this->internal_container) {
+            return;
+        }
+
+        $container = new ContainerBuilder(new ParameterBag($this->container_parameters->all()));
+
+        $this->configureContainerServices($container);
+
+        $container->compile();
+
+        $this->internal_container = $container;
+    }
+
+    private function configureContainerServices(ContainerBuilder $container): void
+    {
+        $configurator = $this->getContainerConfigurator($container);
+
+        // Handy configurator (just like Symfony config files)
+        $services = $configurator->services();
+        $services
+            ->defaults()
+                ->autowire()
+                ->autoconfigure()
+            ->instanceof(PublicService::class)->public()
+        ;
+
+
+        // Autoconfig
+        $container->registerForAutoconfiguration(PublicService::class)->setPublic(true);
+
+
+        // Routes loader
+        $route_loader = $services->set(PluginRoutesLoader::class);
+        $route_loader->bind('$env', $this->kernel->getEnvironment());
+        $services->alias('routing.loader', PluginRoutesLoader::class);
+
+
+        // Plugins routing
+        $router = $services->set('glpi_plugin_router', Router::class)->public();
+        $router->bind('$loader', new Reference(PluginRoutesLoader::class));
+        $router->bind('$resource', 'plugins_routes');
```

Comment:
I'd still keep the generator as alias
```suggestion
        $router->bind('$resource', 'plugins_routes');
        $services->alias(UrlGeneratorInterface::class, 'glpi_plugin_router');
```